data/src/main/java/io/gnosis/data/repositories/TransactionRepository.kt

```diff
@@ -76,35 +76,37 @@ class TransactionRepository(
                 SafeRepository.isSettingsMethod(transactionDto.dataDecoded?.method)
 
     // This is a big assumption for txType == ETHEREUM_TRANSACTION, it was agreed that this can be assumed successful, because only successful TXs trigger events
-    private fun transfer(transferDto: TransferDto): Transaction.Transfer {
-        val tokenInfo = serviceTokenInfo(transferDto)
+    private fun transfer(transfer: TransferDto, safeInfo: SafeInfo): Transaction.Transfer {
+        val tokenInfo = serviceTokenInfo(transfer)
         val value = when (tokenInfo.type) {
             ServiceTokenInfo.TokenType.ERC721 -> BigInteger.ONE
-            else -> transferDto.value ?: BigInteger.ZERO
+            else -> transfer.value ?: BigInteger.ZERO
         }
         return Transaction.Transfer(
             TransactionStatus.Success,
             null,
-            transferDto.to,
-            transferDto.from,
+            transfer.to,
+            transfer.from,
             value,
-            transferDto.executionDate?.formatBackendDate(),
+            transfer.executionDate?.formatBackendDate(),
             tokenInfo,
-            null
+            null,
+            transfer.isIncoming(safeInfo)
         )
     }
 
     // This is a big assumption for txType == ETHEREUM_TRANSACTION, it was agreed that this can be assumed successful, because only successful TXs trigger events
-    private fun transfer(transaction: EthereumTransactionDto): Transaction.Transfer =
+    private fun transfer(ethereumTransaction: EthereumTransactionDto, safeInfo: SafeInfo): Transaction.Transfer =
         Transaction.Transfer(
             TransactionStatus.Success,
             null,
-            transaction.to,
-            transaction.from,
-            transaction.value ?: BigInteger.ZERO,
-            transaction.blockTimestamp?.formatBackendDate(),
+            ethereumTransaction.to,
+            ethereumTransaction.from,
+            ethereumTransaction.value ?: BigInteger.ZERO,
+            ethereumTransaction.blockTimestamp?.formatBackendDate(),
             ETH_SERVICE_TOKEN_INFO,
-            null
+            null,
+            ethereumTransaction.isIncoming(safeInfo)
```

Comment:
This is weird, aren't ethereum transactions generally incoming always? Maybe it is my misunderstanding. I think we should avoid the coupling with `safeInfo` here, just like for the other case.